### PR TITLE
Feature/cvsb 5211

### PR DESCRIPTION
**Pull request by bsiddappa**
Please review this along with PR created for cert-gen repo.

### CODE REVIEW

File: src/main/java/uk/gov/dvsa/PdfGenerator.java [hunk 1]
```diff
@@ -48,14 +48,17 @@ public ApiGatewayResponse handleRequest(Map<String, Object> input, Context conte
...
         try {
-            Map<String, String> requestHeaders = requestParser.getTracingHeaders(input);
-            requestTracingService.setCurrentTracingInformation(
-                    requestHeaders.getOrDefault(REQUEST_TRACE_ID_HEADER, requestTracingService.generateUUID()),
-                    requestHeaders.getOrDefault(REQUEST_SPAN_ID_HEADER, requestTracingService.generateUUID()),
-                    requestHeaders.get(REQUEST_PARENT_SPAN_ID_HEADER)
-            );
-
-            eventLogger.logEventWithTraceInfo(EventType.CERT_SERVER_RECEIVE, requestTracingService.getCurrentTracingInformation());
+            if (input.containsKey(requestParser.REQUEST_HEADERS)) {
```
Comment (oalexandru) [inline src/main/java/uk/gov/dvsa/PdfGenerator.java]: what's the logic behind this? why was this change necessary?

Comment (bsiddappa) [reply src/main/java/uk/gov/dvsa/PdfGenerator.java]: As mentioned, Not changing the existing flow/logic as it may be required in future if we expose it for an api call. Also i did not see this as a mandatory field in the requestParser which would cause any issues, so made this optional here as well.
